DEVELOPMENT.md

```diff
@@ -1,32 +1,108 @@
+### Requirements:
 
-In order to use an independently built or installed version of `terraform-ls`, you will need to set `terraform.languageServer.pathToBinary` to the correct executable path.
+- VSCode >= 1.61
+- Node >= 16.13.2
+- npm >= 8.x
 
-The extension can be run in a development mode (useful for testing syntax highlighting changes, for example) via the debug task called `Launch Client`. This will open a new window with the extension loaded, and from there you can open any files or folders you want to check against. This extension development window can also be used to run commands, and use the language server installer.
+### Getting the code
 
-To package your local development copy for testing, use the [vsce](https://www.npmjs.com/package/vsce) tool with `vcse package`.
+```
+git clone https://github.com/hashicorp/vscode-terraform
+```
 
-# Development
+### Dependencies:
 
-We are an open source project on GitHub and would enjoy your contributions! Please [open a new issue](https://github.com/hashicorp/terraform-vscode-extension/issues) before working on a PR that requires significant effort. This will allow us to make sure the work is in line with the project's goals.
+After cloning the repo, run `npm install` to install dependencies. There's an included build task to compile the TypeScript files to JavaScript; you can run it directly with `npm run compile`.
+
+```
+> npm install
+> npm run compile
+```
+
+> In order to use an independently built or installed version of terraform-ls, you will need to set `terraform.languageServer.pathToBinary` to the correct executable path.
+
+## Running the Extension
+
+The extension can be run in a development mode via the launch task called `Launch Client`. This will open a new VS Code window with the extension loaded, and from there you can open any files or folders you want to check against. This extension development window can also be used to run commands or any other feature the extension provides.
+
+> New to VS Code development? You can get started [here](https://code.visualstudio.com/api/get-started/your-first-extension)
+
+## Tests
+
+Automated `unit` and `integration` tests can be written using [mocha](https://mochajs.org) and live inside `./src/test` with file pattern `*.test.ts`.
+
+>It is *required* that `terraform` is available on `$PATH` to run the tests.
+
+To run the `unit tests` from the command-line run:
+
+```bash
+> `npm test:unit`
+```
+
+To run the `integration tests` from the command-line without any configuration, close any open VSCode windows and run `npm test`. By default, `npm test` will test against VS Code Stable. If you want to test against a different VS Code version, or if you want to have VS Code remain open, use an environment variable to indicate which version of VS Code to test against:
```

Comment:
With the updated version of `@vscode/test-electron` VSCode can remain open, even if you're running tests against stable 🎉